### PR TITLE
(DEV-702) adds playlist endpoint

### PR DESCRIPTION
**Pull request by revdan**
#### Summary
This PR adds the playlist CRUD endpoints.

#### Intended effect
A user should be able to create playlists from categories, and also list/show/edit/delete.

#### How I tested
The new integration tests in `/examples/playlist` do a good job of testing the new functionality. Aside from that, manual testing with Postman.

[Run in Postman](https://app.getpostman.com/run-collection/da7826f2595e845689a1)

NB you will need to make sure you have the `api_envs.yml` file completed to run the integration tests. Let me know if you need help with this.

### CODE REVIEW

File: spec/vzaar_api/playlist_spec.rb [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            expect(playlist.dimensions).to eq '768x340'
+            expect(playlist.position).to eq 'right'
+            expect(playlist.autoplay).to be_truthy
+            expect(playlist.continuous_play).to be_truthy
+            expect(playlist.embed_code).to match(
+              "<iframe id=\"vzpl-12692\" name=\"vzpl-12692\" " \
+              "title=\"vzaar video player\" class=\"vzaar video player\" " \
+              "type=\"text/html\" width=\"927\" height=\"340\" frameborder=\"0\" " \
+              "allowFullScreen allowTransparency=\"true\" mozallowfullscreen " \
+              "webkitAllowFullScreen src=\"//view.vzaar.localhost/playlists/12692\">" \
```
Comment (lawrencesears) [inline spec/vzaar_api/playlist_spec.rb]: This should probably be getting `hostname` from the envs.

File: examples/playlist/create_update_delete_spec.rb [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        expect(playlist.dimensions).to eq '400x200'
+        expect(playlist.position).to eq 'top'
+        expect(playlist.autoplay).to be_truthy
+        expect(playlist.continuous_play).to be_truthy
+        expect(playlist.embed_code).to match(
+          "<iframe id=\"vzpl-#{playlist.id}\" name=\"vzpl-#{playlist.id}\" " \
+          "title=\"vzaar video player\" class=\"vzaar video player\" " \
+          "type=\"text/html\" width=\"400\" height=\"330\" frameborder=\"0\" " \
+          "allowFullScreen allowTransparency=\"true\" mozallowfullscreen " \
+          "webkitAllowFullScreen src=\"//view.vzaar.localhost/playlists/#{playlist.id}\">" \
```
Comment (lawrencesears) [inline examples/playlist/create_update_delete_spec.rb]: Same here.

File: examples/playlist/create_update_delete_spec.rb [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        expect(playlist.dimensions).to eq 'auto'
+        expect(playlist.position).to eq 'left'
+        expect(playlist.autoplay).to be_falsey
+        expect(playlist.continuous_play).to be_falsey
+        expect(playlist.embed_code).to match(
+          "<iframe id=\"vzpl-#{playlist.id}\" name=\"vzpl-#{playlist.id}\" " \
+          "title=\"vzaar video player\" class=\"vzaar video player\" " \
+          "type=\"text/html\" width=\"927\" height=\"432\" frameborder=\"0\" " \
+          "allowFullScreen allowTransparency=\"true\" mozallowfullscreen " \
+          "webkitAllowFullScreen src=\"//view.vzaar.localhost/playlists/#{playlist.id}\">" \
```
Comment (lawrencesears) [inline examples/playlist/create_update_delete_spec.rb]: And here. (I've just done that totally backwards)

File: spec/vzaar_api/playlist_spec.rb (hunk 1 above)
Comment (revdan) [inline spec/vzaar_api/playlist_spec.rb]: 👍

**Pull request merged**